pkg/database/database.go

```diff
@@ -5,408 +5,103 @@ package database
-	if err != nil {
-		return "", "", err
-	}
-	if len(logUrls) >= 1 {
-		return logUrls[0].LogURL, logUrls[0].JsonPath, nil
-	} else {
-		return "", "", ResourceNotFoundError
-	}
-}
-
-func (db *Database) QueryLogURL(ctx context.Context, kind, apiVersion, namespace, name string) (string, string, error) {
-	if kind == "Pod" {
-		sb := db.Selector.ResourceSelector()
-		sb = db.Sorter.CreationTSAndIDSorter(sb) // If resources are named the same, select the newest
-		sb.Where(
-			db.Filter.KindFilter(sb.Cond, kind),
-			db.Filter.ApiVersionFilter(sb.Cond, apiVersion),
-			db.Filter.NamespaceFilter(sb.Cond, namespace),
-			db.Filter.NameFilter(sb.Cond, name),
-		)
-
-		return db.getLogsForPodSelector(ctx, sb, namespace, name)
-	}
-
-	// Not a Pod
-	sb := db.Selector.UUIDResourceSelector()
-	sb = db.Sorter.CreationTSAndIDSorter(sb) // If resources are named the same, select the newest
-	sb.Where(
-		db.Filter.KindFilter(sb.Cond, kind),
-		db.Filter.ApiVersionFilter(sb.Cond, apiVersion),
-		db.Filter.NamespaceFilter(sb.Cond, namespace),
-		db.Filter.NameFilter(sb.Cond, name),
-	)
-
-	strQueryPerformer := newQueryPerformer[string](db.DB, db.Flavor)
-	uuid, err := strQueryPerformer.performSingleRowQuery(ctx, sb)
-	if errors.Is(err, sql.ErrNoRows) {
-		return "", "", ResourceNotFoundError
-	}
-
-	if err != nil {
-		return "", "", err
-	}
-
-	slog.Debug(
-		"getting owned pods for resource",
-		"kind", kind,
-		"namespace", namespace,
-		"name", name,
-		"uuid", uuid,
-	)
-	pods, err := db.getOwnedPodsUuids(ctx, []string{uuid}, []uuidKindDate{})
-	if err != nil {
-		return "", "", err
-	}
-	if len(pods) == 0 {
-		return "", "", ResourceNotFoundError
-	}
-
-	// Get the most recent pod from owned by the provided resource
-	slices.SortFunc(pods, func(a, b uuidKindDate) int {
-		return strings.Compare(b.Date, a.Date)
-	})
-
-	sb = db.Selector.ResourceSelector()
-	sb.Where(db.Filter.UuidFilter(sb.Cond, pods[0].Uuid))
-
-	return db.getLogsForPodSelector(ctx, sb, namespace, name)
+func (db *DatabaseImpl) CloseDB() error {
+	return db.conn.Close()
 }
 
-func (db *Database) getOwnedPodsUuids(ctx context.Context, ownersUuids []string, podUuids []uuidKindDate,
-) ([]uuidKindDate, error) {
-
-	if len(ownersUuids) == 0 {
-		return podUuids, nil
-	} else {
-		sb := db.Selector.OwnedResourceSelector()
-		sb.Where(db.Filter.OwnerFilter(sb.Cond, ownersUuids))
-		parsedRows, err := newQueryPerformer[uuidKindDate](db.DB, db.Flavor).performQuery(ctx, sb)
-		if err != nil {
-			return nil, err
-		}
-		ownersUuids = make([]string, 0)
-		for _, row := range parsedRows {
-			if row.Kind == "Pod" {
-				podUuids = append(podUuids, row)
-			} else {
-				ownersUuids = append(ownersUuids, row.Uuid)
+var once sync.Once
+var singleDatabase Database
+
+// getSingleDatabase implements singleton pattern to prevent executing getDatabaseByEnv more than once
+func getSingleDatabase() (Database, error) {
+	var err error
+	if singleDatabase == nil {
+		once.Do(func() {
```

Comment:
I think these two lines are redundant. `sync.Once` already keeps track of when it was called, so there is no need to check if `singleDatabase` is `nil`. I would remove the variable `once` or remove the check against `nil`.